recipes/boost/all/conanfile.py

```diff
@@ -125,6 +127,34 @@ def configure(self):
         if not self.options.i18n_backend and not self.options.without_locale:
             raise ConanInvalidConfiguration("Boost 'locale' library requires a i18n_backend, either 'icu' or 'iconv'")
 
+        if self.options.threading == "single":
```

Comment:
this might be significantly simplified:
```
for lib in ['locale', 'coroutune', 'wave', ...]:
    if not self.options.get_safe('without_%s' % lib):
        raise ConanInvalidConfiguration("Boost '%s' library requires multi threading" %s)
```